### PR TITLE
Fix Python break due to Boost 1.63

### PR DESCRIPTION
**Pull request by chadwalters**
Boost.Python 1.63 changed the interface to boost::python::converter::shared_ptr_from_python<>.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -27,6 +27,11 @@
               BOND_VS_NUM: 14
               BOND_ARCH: 32
               BOND_BOOST: 60
+            - BOND_BUILD: Python
+              BOND_VS: "Visual Studio 14 2015"
+              BOND_VS_NUM: 14
+              BOND_ARCH: 32
```
Comment (chwarr) [inline appveyor.yml]: We've got a 32-bit variant already. Do you want to use 64 here for increased coverage?

File: python/inc/bond/python/converters.h [hunk 2]
```diff
@@ -458,8 +458,12 @@ inline void register_builtin_converters()
             true
             >();
 
-        // Allows extraction of share_ptr<char> from python string object
-        converter::shared_ptr_from_python<char>();
+        // Allows extraction of shared_ptr<char> from python string object
+#if BOOST_VERSION >= 106300
+        converter::shared_ptr_from_python<char, boost::shared_ptr>();
+#else
+		converter::shared_ptr_from_python<char>();
```
Comment (chwarr) [inline python/inc/bond/python/converters.h]: Minor: the tabs in line 465 are causing odd indentation. Please convert them to spaces.

File: appveyor.yml [hunk 3]
```diff
@@ -27,6 +27,11 @@
               BOND_VS_NUM: 14
               BOND_ARCH: 32
               BOND_BOOST: 60
+            - BOND_BUILD: Python
+              BOND_VS: "Visual Studio 14 2015"
+              BOND_VS_NUM: 14
+              BOND_ARCH: 64
+              BOND_BOOST: 63
```
Comment (chwarr) [inline appveyor.yml]: This build failed in AppVeyor: https://ci.appveyor.com/project/sapek/bond/build/1.0.1272/job/sp5n8a3a06id2twy

File: appveyor.yml [hunk 4]
```diff
@@ -140,11 +145,6 @@
 
                 cd build
 
-                # Make sure we have Python27-64 before any other version; Python path per https://www.appveyor.com/docs/installed-software#python
```
Comment (chwarr) [inline appveyor.yml]: C++ build is failing now. Looks like we may need this in a common locations for 64-bit builds.
https://ci.appveyor.com/project/sapek/bond/build/1.0.1277/job/r229e9ypw9600t8b

Comment (chadwalters) [inline appveyor.yml]: Yep, looks like both Python and the C++ builds need this in their 64-bit config. Spinning a new version

**Comment by chadwalters** [comment]
Looks like this is ready to go. Please Rebase and Merge (not squash).

**Comment by chwarr** [comment]
7a3d4b02 is this rebased atop master.

**Pull request closed without merging**